### ISSUE TITLE
[Feature request] : dedicated nodes per project

### ISSUE THREAD

**Issue by ludopaquet** [rancher/rancher#19291]
**What kind of request is this (question/bug/enhancement/feature request):**

feature request

Hi,

Trying to migrate from Rancher1 to Rancher2, I'd like to avoid creating a cluster per group of virtualmachines (cause in my company we have a set of machines per team). 

I think it could be great to have dedicated nodes per project like Environnement in 1.6. 
It could use under the hood some taints or nodeselector mechanism for every namespaces inside a project.

What do you think ?

**Comment by cjellick** [comment]
What is your need/motivation for hosts per project (beyond it just being how 1.6 behaved)?

**Comment by vincent99** [comment]
It's really the opposite of how 1.x behaved, each environment was a dedicated cluster and there was no way to share physical resources between them.

2.x separates out the clusters of physical resources and the RBAC of who is allowed to use them.  So you can have multiple projects sharing one, or if you don't want sharing, make one cluster per project.

I imagine the real concern is you want "dedicated" resources for each project, but don't really want to pay for (or manage) multiple clusters/etcd/controlplanes. But node scheduling is just one of many concerns that you'd want to address in a sorta-shared cluster.

**Comment by ludopaquet** [comment]
Yes you're right, I don't want to instantiate at least 5 or 6 machines to have a dedicated environnement like we had in 1.6.
Regarding the question of @cjellick, it's a matter of billing (easier when each team has their own nodes) + isolation (don't want that a faulty container of a team A impacts team B)

For now I will try to play with the taint mecanism and force team to add it within their workloads inside a project. Maybe it will do the trick?
